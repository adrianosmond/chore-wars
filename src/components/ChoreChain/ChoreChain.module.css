.chain {
  margin-top: var(--spacing-s);
}

.chore {
  display: flex;
  justify-content: space-between;
  padding: var(--spacing-s);
  background-color: var(--color-gray-100);
  border: 1px solid var(--color-border);
  border-top: 0;
}

.chore:first-child {
  border-top: 1px solid var(--color-border);
  border-top-left-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.chore:last-child {
  border-bottom-left-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
}

.remove {
  margin-top: var(--spacing-s);
  text-align: right;
}
