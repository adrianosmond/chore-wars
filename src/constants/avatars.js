export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
