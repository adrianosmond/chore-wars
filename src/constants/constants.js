const TIME_UNIT = 1000 * 60 * 60 * 24; // 1 day
const MIN_CHORE_FREQUENCY = 1;
const MAX_CHORE_FREQUENCY = 365;
const MIN_CHORE_POINTS = 1;
const MAX_CHORE_POINTS = 200;

const DATE_FORMAT = 'DD/MM/YYYY';

const MAX_POINT_DIFFERENCE = 500;

const MAX_NAME_LENGTH = 10;
const JOIN_CODE_LENGTH = 8;

const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};

const ActionTypes = {
  addChore: 'ADD_CHORE',
  resetChoreDoneDate: 'RESET_CHORE_DONE_DATE',
  blockChore: 'BLOCK_CHORE',
  unblockChore: 'UNBLOCK_CHORE',
  removeChore: 'REMOVE_CHORE',
  updateChore: 'UPDATE_CHORE',
  setChores: 'SET_CHORES',
  makeChain: 'MAKE_CHAIN',
  breakChain: 'BREAK_CHAIN',
  addPoints: 'ADD_POINTS',
  claimPrize: 'CLAIM_PRIZE',
  paidDebt: 'PAID_DEBT',
  setPoints: 'SET_POINTS',
  updateUser: 'UPDATE_USER',
  setAuthUser: 'SET_AUTH_USER',
  setGame: 'SET_GAME',
  setPointsLoaded: 'SET_POINTS_LOADED',
  setChoresLoaded: 'SET_CHORES_LOADED',
  saveStatePostUndo: 'SAVE_STATE_POST_UNDO',
};

export {
  TIME_UNIT,
  MIN_CHORE_FREQUENCY,
  MAX_CHORE_FREQUENCY,
  MIN_CHORE_POINTS,
  MAX_CHORE_POINTS,
  MAX_NAME_LENGTH,
  JOIN_CODE_LENGTH,
  DATE_FORMAT,
  MAX_POINT_DIFFERENCE,
  ActionTypes,
  DefaultAvatar,
};
