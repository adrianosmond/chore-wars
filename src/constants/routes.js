export const LOGIN = '/'
export const CHORES = '/chores'
export const NEW_CHORE = '/chores/new'
