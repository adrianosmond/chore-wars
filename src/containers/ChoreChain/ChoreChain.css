.chore-chain {
  max-width: 600px;
  margin-left: auto;
  margin-right: auto;
}

.chore-chain__title {
  margin-top: 0;
}

.chore-chain__options {
  margin-top: 2rem;
  padding-left: 0;
  list-style-type: none;
}

.chore-chain__option {
  position: relative;
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin-top: 1rem;
  padding: 1rem;
  background-color: white;
}

.chore-chain__checkbox {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  pointer-events: none;
}

.chore-chain__status {
  position: relative;
  padding: 2px;
  height: 1rem;
  width: 1rem;
  min-width: 1rem;
  margin-left: 1rem;
  border: 1px solid navy;
}

.chore-chain__status-icon {
  position: absolute;
  top: 2px;
  bottom: 2px;
  left: 2px;
  right: 2px;
  opacity: 0;
}

.chore-chain__checkbox:checked ~ .chore-chain__status .chore-chain__status-icon {
  opacity: 1;
}

.chore-chain__sort-buttons {
  display: flex;
}

.chore-chain__sort-button {
  margin: 0;
  margin-left: 1rem;
  padding: 0;
  border: 0;
  border-bottom: 2px solid rgba(0,0,0,0.3);
  width: 2.5rem;
  height: 2.5rem;
  border-radius: 0.25rem;
  background-color: navy;
  color: white;
  font-size: 1.5rem;
}

.chore-chain__sort-button:disabled {
  border-top: 2px solid transparent;
  border-bottom-width: 0;
}

.chore-chain__sort-button:disabled {
  opacity: 0.5;
}