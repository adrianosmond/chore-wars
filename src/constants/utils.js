import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
