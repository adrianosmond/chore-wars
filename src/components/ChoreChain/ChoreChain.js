import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import FlipMove from 'react-flip-move';

import * as routes from 'constants/routes';

import checkIcon from 'images/check.svg';
import './ChoreChain.css';

const ChainSelect = props => (
  <div className="chore-chain__section">
    <p>Select all the chores that are part of this chain</p>
    <ul className="chore-chain__options">
      {props.chores.map(chore =>
      <li key={chore.slug}>
        <label className="chore-chain__option">
          <input className="chore-chain__checkbox" type="checkbox" name={chore.slug}
            onChange={() => props.chooseChore(chore)}
            checked={props.chain.findIndex(item => item.slug === chore.slug) >= 0} />
          <span className="chore-chain__chore-title">{chore.title}</span>
          <div className="chore-chain__status">
            <img className="chore-chain__status-icon" src={checkIcon} alt="Include chore in chain" />
          </div>
        </label>
      </li>)}
    </ul>
    <div className="form__button-holder">
      <Link to={routes.CHORES} className="form__button form__button--secondary">Cancel</Link>
      <button disabled={!props.canSave} className="form__button"
        onClick={props.nextStage}>Next</button>
    </div>
  </div>
);

const ChainOrder = props => (
  <div className="chore-chain__section">
    <p>Choose the order that the chores should be done in</p>
    <ul className="chore-chain__options">
      <FlipMove>
      {props.chain.map((chore, idx) =>
      <li key={chore.slug} className="chore-chain__option">
        <span className="chore-chain__chore-title">{chore.title}</span>
        <div className="chore-chain__sort-buttons">
          <button className="chore-chain__sort-button" onClick={() => props.moveChoreUp(idx)}
            disabled={idx === 0}>&uarr;</button>
          <button className="chore-chain__sort-button" onClick={() => props.moveChoreDown(idx)}
            disabled={idx === props.chain.length - 1}>&darr;</button>
        </div>
      </li>)}
      </FlipMove>
    </ul>
    <div className="form__button-holder">
      <button className="form__button form__button--secondary"
        onClick={props.prevStage}>Back</button>
      <button disabled={!props.canSave} className="form__button"
        onClick={() => props.saveChain(props.chain)}>Save</button>
    </div>
  </div>
);

class ChoreChain extends Component {
  constructor(props) {
    super(props);
    this.state = {
      chain: [],
      stage: 'selection',
    };
  }

  chooseChore(chosenChore) {
    let chain = this.state.chain.slice();
    const newChain = chain.filter(chore => chore.slug !== chosenChore.slug);
    if (chain.length === newChain.length) {
      chain.push(chosenChore);
    } else {
      chain = newChain;
    }
    this.setState({
      chain,
    });
  }

  moveChoreUp(idx) {
    const chain = this.state.chain.slice();
    if (chain.length === 1 || idx === 0) return;
    [chain[idx], chain[idx - 1]] = [chain[idx - 1], chain[idx]];
    this.setState({
      chain,
    });
  }

  moveChoreDown(idx) {
    const chain = this.state.chain.slice();
    if (chain.length === 1 || idx === chain.length - 1) return;
    [chain[idx], chain[idx + 1]] = [chain[idx + 1], chain[idx]];
    this.setState({
      chain,
    });
  }

  render() {
    const { chores, saveChain } = this.props;
    if (!chores) return null;
    const { chain, stage } = this.state;
    const canSave = chain.length > 1;
    return (
      <div className="chore-chain">
        <h1 className="chore-chain__title">Create a Chain</h1>
        { stage === 'selection' ?
          <ChainSelect chores={chores} chain={chain} canSave={canSave}
            chooseChore={this.chooseChore.bind(this)}
            nextStage={() => { this.setState({ stage: 'sorting' }); }} />
        : null }
        { stage === 'sorting' ?
          <ChainOrder chain={chain} canSave={canSave}
            moveChoreUp={this.moveChoreUp.bind(this)}
            moveChoreDown={this.moveChoreDown.bind(this)}
            prevStage={() => { this.setState({ stage: 'selection' }); }}
            saveChain={saveChain} />
        : null }
      </div>
    );
  }
}

export { ChainSelect, ChainOrder };

export default ChoreChain;
