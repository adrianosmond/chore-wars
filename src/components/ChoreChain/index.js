import ChoreChain from './ChoreChain';

export default ChoreChain;
