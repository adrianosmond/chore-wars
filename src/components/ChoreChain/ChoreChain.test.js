import React from 'react';
import { shallow } from 'enzyme';
import { filterAndSortChores } from 'constants/utils';

import ChoreChain, { ChainSelect, ChainOrder } from './ChoreChain';

const chores = filterAndSortChores({
  first: {
    frequency: 7,
    lastDone: 0,
    pointsPerTime: 50,
    title: 'First',
  },
  second: {
    frequency: 0,
    lastDone: 10,
    pointsPerTime: 10,
    title: 'Second',
  },
  third: {
    frequency: 0,
    lastDone: 0,
    pointsPerTime: 20,
    title: 'Third',
  },
  fourth: {
    frequency: 1,
    lastDone: 100,
    pointsPerTime: 100,
    title: 'Fourth',
  },
});

describe('Chore Chain', () => {
  describe('Class', () => {
    const component = shallow(<ChoreChain chores={chores} />);
    it('Renders Selection', () => {
      expect(component).toMatchSnapshot();
    });


    it('Adds/removes a chore to/from the chain on selection', () => {
      const checkbox = component.find(ChainSelect).dive().find('.chore-chain__checkbox').first();
      checkbox.simulate('change', { target: { checked: true } });
      expect(component.state('chain')).toEqual([chores[0]]);
      checkbox.simulate('change', { target: { checked: true } });
      expect(component.state('chain')).toEqual([]);
    });

    it('Renders Sorting', () => {
      component.setState({
        chain: [chores[0], chores[1]],
        stage: 'sorting',
      });
      expect(component).toMatchSnapshot();
    });

    it('Moves chores up', () => {
      const checkbox = component.find(ChainOrder).dive().find('.chore-chain__sort-button').at(2);
      checkbox.simulate('click');
      expect(component.state('chain')).toEqual([chores[1], chores[0]]);
    });

    it('Moves chores down', () => {
      const checkbox = component.find(ChainOrder).dive().find('.chore-chain__sort-button').at(1);
      checkbox.simulate('click');
      expect(component.state('chain')).toEqual([chores[0], chores[1]]);
    });
  });

  describe('Selection', () => {
    const chooseChore = jest.fn();
    const nextStage = jest.fn();
    const component = shallow(<ChainSelect
      chores={chores}
      chain={[]}
      chooseChore={chooseChore}
      canSave
      nextStage={nextStage}
    />);

    it('Renders', () => {
      expect(component).toMatchSnapshot();
    });

    it('Handles clicks on checkboxes', () => {
      expect(chooseChore).not.toHaveBeenCalled();
      component.find('.chore-chain__checkbox').first().simulate('change', { target: { checked: true } });
      expect(chooseChore).toHaveBeenCalledTimes(1);
    });

    it('Can go to the next stage', () => {
      expect(nextStage).not.toHaveBeenCalled();
      component.find('.form__button').at(1).simulate('click');
      expect(nextStage).toHaveBeenCalledTimes(1);
    });
  });

  describe('Sorting', () => {
    const moveChoreUp = jest.fn();
    const moveChoreDown = jest.fn();
    const prevStage = jest.fn();
    const saveChain = jest.fn();
    const component = shallow(<ChainOrder
      chores={chores}
      chain={[chores[0], chores[1], chores[2]]}
      canSave
      prevStage={prevStage}
      saveChain={saveChain}
      moveChoreDown={moveChoreDown}
      moveChoreUp={moveChoreUp}
    />);

    it('Renders', () => {
      expect(component).toMatchSnapshot();
    });

    it('Can move chores up (but not the first)', () => {
      expect(moveChoreUp).not.toHaveBeenCalled();
      expect(component.find('.chore-chain__sort-button').first().prop('disabled')).toBe(true);
      component.find('.chore-chain__sort-button').at(2).simulate('click');
      expect(moveChoreUp).toHaveBeenCalledTimes(1);
    });

    it('Can move chores down (but not the last)', () => {
      expect(moveChoreDown).not.toHaveBeenCalled();
      component.find('.chore-chain__sort-button').at(1).simulate('click');
      expect(moveChoreDown).toHaveBeenCalledTimes(1);
      expect(component.find('.chore-chain__sort-button').last().prop('disabled')).toBe(true);
    });

    it('Can go to the previous stage', () => {
      expect(prevStage).not.toHaveBeenCalled();
      component.find('.form__button').first().simulate('click');
      expect(prevStage).toHaveBeenCalledTimes(1);
    });

    it('Can save the chain', () => {
      expect(saveChain).not.toHaveBeenCalled();
      component.find('.form__button').at(1).simulate('click');
      expect(saveChain).toHaveBeenCalledTimes(1);
    });
  });
});
