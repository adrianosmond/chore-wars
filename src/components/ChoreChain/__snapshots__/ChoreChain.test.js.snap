// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chore Chain Class Renders Selection 1`] = `
<div
  className="chore-chain"
>
  <h1
    className="chore-chain__title"
  >
    Create a Chain
  </h1>
  <ChainSelect
    canSave={false}
    chain={Array []}
    chooseChore={[Function]}
    chores={
      Array [
        Object {
          "currentPoints": Infinity,
          "due": 604800000,
          "frequency": 7,
          "lastDone": 0,
          "percentage": 100,
          "pointsPerTime": 50,
          "slug": "first",
          "title": "First",
        },
        Object {
          "currentPoints": Infinity,
          "due": 86400100,
          "frequency": 1,
          "lastDone": 100,
          "percentage": 100,
          "pointsPerTime": 100,
          "slug": "fourth",
          "title": "Fourth",
        },
        Object {
          "currentPoints": 10,
          "due": 10,
          "frequency": 0,
          "lastDone": 10,
          "percentage": 100,
          "pointsPerTime": 10,
          "slug": "second",
          "title": "Second",
        },
        Object {
          "currentPoints": 20,
          "due": 0,
          "frequency": 0,
          "lastDone": 0,
          "percentage": 100,
          "pointsPerTime": 20,
          "slug": "third",
          "title": "Third",
        },
      ]
    }
    nextStage={[Function]}
  />
</div>
`;

exports[`Chore Chain Class Renders Sorting 1`] = `
<div
  className="chore-chain"
>
  <h1
    className="chore-chain__title"
  >
    Create a Chain
  </h1>
  <ChainOrder
    canSave={true}
    chain={
      Array [
        Object {
          "currentPoints": Infinity,
          "due": 604800000,
          "frequency": 7,
          "lastDone": 0,
          "percentage": 100,
          "pointsPerTime": 50,
          "slug": "first",
          "title": "First",
        },
        Object {
          "currentPoints": Infinity,
          "due": 86400100,
          "frequency": 1,
          "lastDone": 100,
          "percentage": 100,
          "pointsPerTime": 100,
          "slug": "fourth",
          "title": "Fourth",
        },
      ]
    }
    moveChoreDown={[Function]}
    moveChoreUp={[Function]}
    prevStage={[Function]}
  />
</div>
`;

exports[`Chore Chain Selection Renders 1`] = `
<div
  className="chore-chain__section"
>
  <p>
    Select all the chores that are part of this chain
  </p>
  <ul
    className="chore-chain__options"
  >
    <li
      key="first"
    >
      <label
        className="chore-chain__option"
      >
        <input
          checked={false}
          className="chore-chain__checkbox"
          name="first"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="chore-chain__chore-title"
        >
          First
        </span>
        <div
          className="chore-chain__status"
        >
          <img
            alt="Include chore in chain"
            className="chore-chain__status-icon"
            src="check.svg"
          />
        </div>
      </label>
    </li>
    <li
      key="fourth"
    >
      <label
        className="chore-chain__option"
      >
        <input
          checked={false}
          className="chore-chain__checkbox"
          name="fourth"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="chore-chain__chore-title"
        >
          Fourth
        </span>
        <div
          className="chore-chain__status"
        >
          <img
            alt="Include chore in chain"
            className="chore-chain__status-icon"
            src="check.svg"
          />
        </div>
      </label>
    </li>
    <li
      key="second"
    >
      <label
        className="chore-chain__option"
      >
        <input
          checked={false}
          className="chore-chain__checkbox"
          name="second"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="chore-chain__chore-title"
        >
          Second
        </span>
        <div
          className="chore-chain__status"
        >
          <img
            alt="Include chore in chain"
            className="chore-chain__status-icon"
            src="check.svg"
          />
        </div>
      </label>
    </li>
    <li
      key="third"
    >
      <label
        className="chore-chain__option"
      >
        <input
          checked={false}
          className="chore-chain__checkbox"
          name="third"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="chore-chain__chore-title"
        >
          Third
        </span>
        <div
          className="chore-chain__status"
        >
          <img
            alt="Include chore in chain"
            className="chore-chain__status-icon"
            src="check.svg"
          />
        </div>
      </label>
    </li>
  </ul>
  <div
    className="form__button-holder"
  >
    <Link
      className="form__button form__button--secondary"
      replace={false}
      to="/chores"
    >
      Cancel
    </Link>
    <button
      className="form__button"
      disabled={false}
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Chore Chain Sorting Renders 1`] = `
<div
  className="chore-chain__section"
>
  <p>
    Choose the order that the chores should be done in
  </p>
  <ul
    className="chore-chain__options"
  >
    <FlipMovePropConverter
      delay={0}
      disableAllAnimations={false}
      duration={350}
      easing="ease-in-out"
      enterAnimation="elevator"
      getPosition={[Function]}
      leaveAnimation="elevator"
      maintainContainerHeight={false}
      staggerDelayBy={0}
      staggerDurationBy={0}
      typeName="div"
      verticalAlignment="top"
    >
      <li
        className="chore-chain__option"
        key="first"
      >
        <span
          className="chore-chain__chore-title"
        >
          First
        </span>
        <div
          className="chore-chain__sort-buttons"
        >
          <button
            className="chore-chain__sort-button"
            disabled={true}
            onClick={[Function]}
          >
            ↑
          </button>
          <button
            className="chore-chain__sort-button"
            disabled={false}
            onClick={[Function]}
          >
            ↓
          </button>
        </div>
      </li>
      <li
        className="chore-chain__option"
        key="fourth"
      >
        <span
          className="chore-chain__chore-title"
        >
          Fourth
        </span>
        <div
          className="chore-chain__sort-buttons"
        >
          <button
            className="chore-chain__sort-button"
            disabled={false}
            onClick={[Function]}
          >
            ↑
          </button>
          <button
            className="chore-chain__sort-button"
            disabled={false}
            onClick={[Function]}
          >
            ↓
          </button>
        </div>
      </li>
      <li
        className="chore-chain__option"
        key="second"
      >
        <span
          className="chore-chain__chore-title"
        >
          Second
        </span>
        <div
          className="chore-chain__sort-buttons"
        >
          <button
            className="chore-chain__sort-button"
            disabled={false}
            onClick={[Function]}
          >
            ↑
          </button>
          <button
            className="chore-chain__sort-button"
            disabled={true}
            onClick={[Function]}
          >
            ↓
          </button>
        </div>
      </li>
    </FlipMovePropConverter>
  </ul>
  <div
    className="form__button-holder"
  >
    <button
      className="form__button form__button--secondary"
      onClick={[MockFunction]}
    >
      Back
    </button>
    <button
      className="form__button"
      disabled={false}
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
